tests/src/Integration/Engine/IntegrityCheckTest.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\Tests\rules\Integration\Engine\IntegrityCheckTest.
+ */
+
+namespace Drupal\Tests\rules\Integration\Engine;
+
+use Drupal\rules\Context\ContextConfig;
+use Drupal\rules\Engine\ConfigurationState;
+use Drupal\Tests\rules\Integration\RulesEntityIntegrationTestBase;
+
+/**
+ * Test the integrity check functionality during configuration time.
+ *
+ * @group rules
+ */
+class IntegrityCheckTest extends RulesEntityIntegrationTestBase {
+
+  /**
+   * Tests that the integrity check can be invoked.
+   */
+  public function testIntegrityCheck() {
+    $rule = $this->rulesExpressionManager->createRule();
+    $rule->addAction('rules_entity_save', ContextConfig::create()
+      ->map('entity', 'entity')
+    );
+
+    $config_state = ConfigurationState::create([
```

Comment:
I think we should put this logic into the RulesComponent helper class also, such that you can do
RulesComponent($rule)
  ->checkIntegrity()
  ->execute();

(with execute being optional...)